src/main/java/com/ericsson/ei/frontend/utils/BackendInformationControllerImpl.java

```diff
@@ -0,0 +1,95 @@
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package com.ericsson.ei.frontend.utils;
+
+import java.net.URI;
+import java.util.stream.Collectors;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+
+import com.google.gson.JsonArray;
+
+@Component
+public class BackendInformationControllerImpl {
+
+    private static final Logger LOG = LoggerFactory.getLogger(BackendInformationControllerImpl.class);
+
+    @Autowired
+    private BackendInstancesHandler backendInstancesUtils;
+
+    /**
+     * Processes a request to get all available back ends and returns
+     * data containing all back end instances with added information
+     * weather a back end is set as active or not.
+     *
+     * @param request
+     * @return new ResponseEntity
+     */
+    public ResponseEntity<String> handleRequestForInstances(HttpServletRequest request) {
+        try {
+            JsonArray allAvailableInstances = backendInstancesUtils.getBackendInstancesAsJsonArray();
+
+            return new ResponseEntity<>(
+                    allAvailableInstances.toString(),
+                    getHeaders(), HttpStatus.OK);
+        } catch (Exception e) {
+            LOG.error("ERROR!\n" + e.getMessage());
+            return new ResponseEntity<>(
+                    "{\"message\": \"Failure when trying to load backend instances\"}",
+                    getHeaders(), HttpStatus.NOT_FOUND);
+        }
+    }
+
+    /**
+     * Processes requests to switch back end, and sets the back end name to a session variable that is
+     * kept for current session.
+     *
+     * @param request
+     * @return new ResponseEntity
+     */
+    public ResponseEntity<String> handleRequestToSwitchBackEnd(HttpServletRequest request) {
+        try {
+            String selectedInstanceName = request.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
+            request.getSession().setAttribute("backEndInstanceName", selectedInstanceName);
+
+            return new ResponseEntity<>(
+                    "{\"message\": \"Backend instance with name '" + selectedInstanceName + "' was selected.\"}",
+                    getHeaders(), HttpStatus.OK);
+        } catch (Exception e) {
+            LOG.error("Error while switching instance: " + e.getMessage());
+            String response = "{\"message\": \"Internal Error: " + e.getMessage() + "\"}";
+            return new ResponseEntity<>(response, getHeaders(), HttpStatus.INTERNAL_SERVER_ERROR);
```

Comment:
Will it always be an Internal Server Error? Does the backend need authentication? If so what happens when it get bad credentials or other error messages?

Reply:
I don't know why git says i've changed this method. This is not something i changed or want to change in this PR. I've done too much refactoring :) This change might affect tests as other return statuses would be created.

Reply:
The backend switch only takes place in the frontend and doesn't require authentication. I guess it's just a lazy way of catching any potential error caused.